splunk-otel-android/src/main/java/com/splunk/rum/SlowRenderingDetectorImpl.java

```diff
@@ -0,0 +1,116 @@
+package com.splunk.rum;
+
+import static androidx.core.app.FrameMetricsAggregator.DRAW_DURATION;
+import static androidx.core.app.FrameMetricsAggregator.DRAW_INDEX;
+import static com.splunk.rum.SplunkRum.LOG_TAG;
+
+import android.app.Activity;
+import android.util.Log;
+import android.util.SparseIntArray;
+
+import androidx.core.app.FrameMetricsAggregator;
+
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+
+public class SlowRenderingDetectorImpl implements SlowRenderingDetector {
+
+    public static final int SLOW_THRESHOLD_MS = 16;
```

Comment:
These seem somewhat arbitrary. I suspect people will not realise that frozen doesn't mean completely hung but took 700ms to render. These values should be mentioned somewhere. Maybe even make them configurable?

Reply:
These values are taken directly from the android docs. https://developer.android.com/topic/performance/vitals/frozen
16ms is a threshhold for 60FPS rendering. 700ms is a google definition.

I agree that making these configurable could be useful, but I think that should be a follow-up issue/item.